src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -99,7 +99,7 @@ public Http2Stream(HttpRequestMessage request, Http2Connection connection)
 
                 _responseProtocolState = ResponseProtocolState.ExpectingStatus;
 
-                _responseBuffer = new ArrayBuffer(InitialStreamBufferSize, usePool: true);
+                _responseBuffer = new MultiArrayBuffer(InitialStreamBufferSize);
```

Comment:
I don't know if it's going to matter, but this does mean we're now allocating one or more byte[][]s per HTTP/2 stream, whereas previously all of the arrays here were from the pool.

Reply:
Yes.

We could pool the block array too, if we wanted to; I'm just not sure whether that is worthwhile or not.